### PR TITLE
Docs: New plugins pinned versions APIs

### PR DESCRIPTION
**Pull request by tomhjp**
API docs for the new plugin version pinning API. I haven't planned it yet, but if there's any usage documentation it will follow in a separate PR.

### CODE REVIEW

**Review by thyton — APPROVED** [review]
LGMT

File: website/content/api-docs/system/plugins-pins.mdx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+| Method | Path                            |
+| :----- | :------------------------------ |
+| `POST` | `/sys/plugins/pins/:type/:name` |
+
+### Parameters
+
+- `type` `(string: <required>)` – Specifies the type of plugin to create a pinned
+  version for. May be "auth", "database", or "secret". This is part of the request
+  URL.
```
Comment (schavis) [inline website/content/api-docs/system/plugins-pins.mdx:32]:
```suggestion
- `type` `(string: <required>)` – Path parameter specifying the target plugin
  type for the pinned version. May be "auth", "database", or "secret".
```

Style correction: write in active voice

File: website/content/api-docs/system/plugins-pins.mdx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+- `name` `(string: <required>)` – Specifies the name of the plugin to create a
+  pinned version for. This is part of the request URL.
```
Comment (schavis) [inline website/content/api-docs/system/plugins-pins.mdx:35]:
```suggestion
- `name` `(string: <required>)` – Path parameter specifying the plugin name for
  the pinned version.
```

Style correction : write in active voice

File: website/content/api-docs/system/plugins-pins.mdx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+- `version` `(string: <required>)` - Specifies the semantic version of the plugin
+  to pin. Cannot be a builtin version, and must already exist in the catalog.
```
Comment (schavis) [inline website/content/api-docs/system/plugins-pins.mdx:38]:
```suggestion
- `version` `(string: <required>)` - Specifies the semantic version of the plugin
  to pin. Cannot be a builtin version and must already exist in the catalog.
```

File: website/content/api-docs/system/plugins-pins.mdx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                "version": "v1.0.0"
+            }
+        ]
+    }
+}
+```
+
+## Read pinned version
+
+This endpoint reads the pinned version for a plugin with the given type and name.
```
Comment (schavis) [inline website/content/api-docs/system/plugins-pins.mdx:92]:
```suggestion
Read the pinned version for a plugin with the given type and name.
```

File: website/content/api-docs/system/plugins-pins.mdx [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+| Method | Path                            |
+| :----- | :------------------------------ |
+| `GET`  | `/sys/plugins/pins/:type/:name` |
+
+### Parameters
+
+- `type` `(string: <required>)` – Specifies the type of plugin. May be "auth",
+  "database", or "secret". This is part of the request URL.
```
Comment (schavis) [inline website/content/api-docs/system/plugins-pins.mdx:101]:
```suggestion
- `type` `(string: <required>)` – Path parameter specifying the target plugin
  type for the pinned version. May be "auth", "database", or "secret".
```

Edited to use similar language as the other endpoints

File: website/content/api-docs/system/plugins-pins.mdx [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+- `name` `(string: <required>)` – Specifies the name of the plugin. This is
+  part of the request URL.
```
Comment (schavis) [inline website/content/api-docs/system/plugins-pins.mdx:104]:
```suggestion
- `name` `(string: <required>)` – Path parameter specifying the plugin name for
  the pinned version.
```

File: website/content/api-docs/system/plugins-pins.mdx [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        "name": "github",
+        "type": "auth",
+        "version": "v1.0.0"
+    }
+}
+```
+
+## Delete pinned version
+
+This endpoint deletes any pinned version for a plugin with the given type and name.
```
Comment (schavis) [inline website/content/api-docs/system/plugins-pins.mdx:129]:
```suggestion
Delete any pinned version for a plugin with the given type and name.
```

File: website/content/api-docs/system/plugins-pins.mdx [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+
+| Method   | Path                           |
+| :------- | :----------------------------- |
+| `DELETE` | `/sys/plugins/pins/:type/:name |
+
+### Parameters
+
+- `type` `(string: <required>)` – Specifies the type of plugin. May be "auth",
+  "database", or "secret". This is part of the request URL.
```
Comment (schavis) [inline website/content/api-docs/system/plugins-pins.mdx:138]:
```suggestion
- `type` `(string: <required>)` – Path parameter specifying the target plugin
  type for the pinned version. May be "auth", "database", or "secret".
```

File: website/content/api-docs/system/plugins-pins.mdx [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+- `name` `(string: <required>)` – Specifies the name of the plugin. This is
+  part of the request URL.
```
Comment (schavis) [inline website/content/api-docs/system/plugins-pins.mdx:141]:
```suggestion
- `name` `(string: <required>)` – Path parameter specifying the plugin name for
  the pinned version.
```

**Review by schavis — CHANGES_REQUESTED** [review]

**Review by yhyakuna — APPROVED** [review]
LGTM

**Review by schavis — APPROVED** [review]

**Comment by tomhjp** [comment]
Thanks!

**Pull request merged**